src/main/java/seedu/address/logic/commands/ApplicationEditCommand.java

```diff
@@ -0,0 +1,215 @@
+package seedu.address.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_COMPANY;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_CONTACT;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_DATE;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_EMAIL;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_POSITION;
+import static seedu.address.model.ApplicationModel.PREDICATE_SHOW_ALL_APPLICATIONS;
+
+import java.util.List;
+import java.util.Optional;
+
+import seedu.address.commons.core.Messages;
+import seedu.address.commons.core.index.Index;
+import seedu.address.commons.util.CollectionUtil;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.ApplicationModel;
+import seedu.address.model.application.Application;
+import seedu.address.model.application.Company;
+import seedu.address.model.application.Contact;
+import seedu.address.model.application.Date;
+import seedu.address.model.application.Email;
+import seedu.address.model.application.Position;
+
+/**
+ * Edits the details of an existing application in the application book.
+ */
+public class ApplicationEditCommand extends ApplicationCommand {
+
+    public static final String COMMAND_WORD = "edit";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Edits the details of application identified "
+            + "by the index number used in the displayed application list. "
+            + "Existing values will be overwritten by the input values.\n"
+            + "Parameters: INDEX (must be a positive integer) "
+            + "[" + PREFIX_COMPANY + "COMPANY] "
+            + "[" + PREFIX_CONTACT + "CONTACT] "
+            + "[" + PREFIX_EMAIL + "EMAIL] "
+            + "[" + PREFIX_POSITION + "POSITION] "
+            + "[" + PREFIX_DATE + "DATE]..\n"
+            + "Example: " + COMMAND_WORD + " 1 "
+            + PREFIX_CONTACT + "91234567 "
+            + PREFIX_EMAIL + "johndoe@example.com";
```

Comment:
not extremely crucial but maybe can change the example email to be more in-line with the context? 
e.g. internships@shopee.com so that it aligns with the sample application and typical application classes